pkg/measure-config/src/measurefs.go

```diff
@@ -0,0 +1,278 @@
+package main
+
+import (
+	"crypto/sha256"
+	"encoding/hex"
+	"fmt"
+	"io"
+	"log"
+	"os"
+	"path/filepath"
+	"sort"
+
+	"github.com/google/go-tpm/tpm2"
+	"github.com/google/go-tpm/tpmutil"
+)
+
+const (
+	//TpmDevicePath is the TPM device file path
+	TpmDevicePath   = "/dev/tpmrm0"
+	ConfigPCRIndex  = 14
+	ConfigPCRHandle = tpmutil.Handle(tpm2.PCRFirst + ConfigPCRIndex)
+)
+
+type FileInfo struct {
+	exist          bool
+	measureContent bool
+}
+
+type TpmEvent struct {
+	data string
+	pcr  []byte
+}
+
+// we do not measure content of following files
+// because they are unique for each device
+func getExcludeList() []string {
+	return []string{
+		"/config/tpm_credential",
+		"/config/device.cert.pem",
+		"/config/device.key.pem",
+		"/config/onboard.cert.pem",
+		"/config/onboard.key.pem",
+		"/config/soft_serial",
+	}
+}
+
+func isInExcludeList(path string) bool {
+	for _, file := range getExcludeList() {
+		if file == path {
+			return true
+		}
+	}
+	return false
+}
+
+// theses file may appear later on the device and we record the
```

Comment:
How can they appear later if the config partition is now mounted read-only?

Reply:
The attacker can do it using different methods. e.g boot from live USB or just pull the drive from a device

Reply:
I see, thanks.